release-automations/create_release_branch.py

```diff
@@ -0,0 +1,527 @@
+            {'selector': 'tr td:last-child', 'props': [('white-space', 'nowrap')]},
+            {'selector': 'tr:hover', 'props': [('background-color', '#bfeaf9')]},
+        ]
+
+        html_sections = []
+
+        if successful_components:
+            # Convert successful dict to a simpler table
+            s_df = pd.DataFrame.from_dict(successful_components, orient='index').reset_index()
+            # Expect columns: index, url, commit, branch
+            s_df.columns = ["Component", "URL", "Commit", "Branch"]
+            s_styled = s_df.style.set_table_attributes('border="1" class="dataframe table table-hover table-bordered"').set_table_styles(styles).hide()
+            html_sections.append(f"<p><b>Successful Components</b></p>{s_styled.to_html()}")
+
+        if failed_components:
+            f_df = pd.DataFrame.from_dict(failed_components, orient='index').reset_index()
+            f_df.columns = ["Component", "Error"]
+            # Add a Status column for coloring
+            f_df['Status'] = 'Failure'
+            # Move Status to second column
+            f_df = f_df[["Component", "Status", "Error"]]
+            f_styled = f_df.style.apply(self.highlightCells, subset=['Status'], axis=0).set_table_attributes('border="1" class="dataframe table table-hover table-bordered"').set_table_styles(styles).hide()
+            html_sections.append(f"<p><b>Failed Components</b></p>{f_styled.to_html()}")
+
+        # Validation table: always most important
+        if validation_components:
+            v_df = pd.DataFrame.from_dict(validation_components, orient='index').reset_index()
+            v_df.columns = ["Component", "Status", "Plan Commit", "Branch Commit"]
+            v_styled = v_df.style.apply(self.highlightCells, subset=['Status'], axis=0).set_table_attributes('border="1" class="dataframe table table-hover table-bordered"').set_table_styles(styles).hide()
+            html_sections.append(f"<p><b>Validation Table</b></p>{v_styled.to_html()}")
+        
+        header_msg = "Below is the branching status and commit validation for various Rock components:"  # placeholder
+        html_body = "".join(html_sections) if html_sections else "<p>No components processed.</p>"
+        html = f"<!DOCTYPE html><html><head></head><h4>{header_msg}</h4>{html_body}</html>"
+
+        sender_email = "jenkins-compute@amd.com"
+        recipients = [email.strip() for email in MAILING_LIST.split(',') if email.strip()]
+        msg = MIMEMultipart('alternative')
+        msg['Subject'] = f"Branching Status Mail - {self.environment}".strip()
+        msg['From'] = sender_email
+        msg['To'] = ", ".join(recipients)
+        part1 = MIMEText(html, "html")
+        msg.attach(part1)
+        try:
+            s = smtplib.SMTP('torsmtp10.amd.com')
+            s.send_message(msg)
+            s.quit()
+        except Exception as e:
+            logging.error("Failed to send email: %s", e)
```

Comment:
Can we leave the email notification code out of the initial version of this script? I'm not sure if this is the right "sender_email" to be putting in public github code and I think logging and monitoring should be considered after we have the mechanics in place.